collector/generic_collector.go

```diff
@@ -83,12 +87,18 @@ func NewCollector(collectorName string, configFile []byte) (*GenericCollector, e
 		minPollFrequency = minSupportedFrequency
 	}
 
+	if len(configInJSON.MetricsConfig) > metricCountLimit {
+		return nil, fmt.Errorf("Too many metrics defined: %d limit: %d", len(configInJSON.MetricsConfig), metricCountLimit)
```

Comment:
IIUC, this check will fail monitoring completely for a container. Is that the behavior that we want?
Would it be better to just fail collection of user defined metrics?